tfhe/src/core_crypto/gpu/entities/lwe_packing_keyswitch_key.rs

```diff
@@ -0,0 +1,84 @@
+use crate::core_crypto::gpu::vec::CudaVec;
+use crate::core_crypto::gpu::{convert_lwe_keyswitch_key_async, CudaStreams};
+use crate::core_crypto::prelude::{
+    lwe_packing_keyswitch_key_input_key_element_encrypted_size, CiphertextModulus,
+    DecompositionBaseLog, DecompositionLevelCount, GlweSize, LweDimension,
+    LwePackingKeyswitchKeyOwned, PolynomialSize, UnsignedInteger,
+};
+
+#[derive(Debug)]
+pub struct CudaLwePackingKeyswitchKey<T: UnsignedInteger> {
+    pub(crate) d_vec: CudaVec<T>,
+    decomp_base_log: DecompositionBaseLog,
+    decomp_level_count: DecompositionLevelCount,
+    output_glwe_size: GlweSize,
+    output_polynomial_size: PolynomialSize,
+    ciphertext_modulus: CiphertextModulus<T>,
+}
+
+impl<T: UnsignedInteger> CudaLwePackingKeyswitchKey<T> {
+    pub fn from_lwe_packing_keyswitch_key(
+        h_ksk: &LwePackingKeyswitchKeyOwned<T>,
+        streams: &CudaStreams,
+    ) -> Self {
+        let decomp_base_log = h_ksk.decomposition_base_log();
+        let decomp_level_count = h_ksk.decomposition_level_count();
+        let input_lwe_size = h_ksk.input_key_lwe_dimension().to_lwe_size();
+        let output_glwe_size = h_ksk.output_key_glwe_dimension().to_glwe_size();
+        let output_polynomial_size = h_ksk.output_polynomial_size();
+        let ciphertext_modulus = h_ksk.ciphertext_modulus();
+
+        // Allocate memory
+        let mut d_vec = CudaVec::<T>::new_multi_gpu(
+            input_lwe_size.to_lwe_dimension().0
+                * lwe_packing_keyswitch_key_input_key_element_encrypted_size(
+                    decomp_level_count,
+                    output_glwe_size,
+                    output_polynomial_size,
+                ),
```

Comment:
you can likely get the size of the container with input.as_ref().len() to avoid recomputing the size here

Reply:
Nice.